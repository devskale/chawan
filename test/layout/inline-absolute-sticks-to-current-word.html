<!DOCTYPE html>
<!--
This matches neither Blink, nor Gecko, but this is fine:
* Gecko measures absolute height as one line, but we (like Blink) measure it
  as the container's height.  This is an arbitrary decision (implementation
  convenience.)
* Blink doesn't flush EOL absolutes, but we (like Gecko, sort of) do.
  This is intentional, as otherwise the link marker pseudo-elements would
  not stick to the links in many cases.
	- Do note however that Gecko keeps absolutes at EOL if they are
	  preceeded by a space.  I don't, because this is more convenient
	  to implement and also seems more consistent.
--->
<style>* { font-family: monospace; font-size: 16px }</style>
<div style="position: absolute">
test
<div style="width: 25ch; text-align: right; background: red; position: relative">
<span style="position: relative">
test
<span>
test  
<span style="position: absolute; height: 100%; background: blue; top: 1em">
where am I?
</span>
</span>
test test test test test
</div>
</div>
</div>
<div align=right style="width: 20ch">flush</div>
<div style="width: 10ch; margin-top: 3em">
<span>test test <span style="position: absolute; color: red">hi</span>test
</div>
<div style="width: 10ch; position: relative">
<span style="position: relative">test test <span style="position: absolute; color: red; top: 0">hi</span></span>test
</div>
<!-- diverges from Gecko: -->
<div style="width: 10ch">
<span>test test <span style="position: absolute; color: red">hi</span> test
</div>
