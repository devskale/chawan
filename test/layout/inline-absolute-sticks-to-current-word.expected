test           flush
[48;2;255;0;0m test test test test test[49m
[48;2;255;0;0m           [48;2;0;0;255mwhere am I?[48;2;255;0;0mest[49m
           [48;2;0;0;255m           [49m
test test
[38;2;255;40;40mhi[39mst
[38;2;255;40;40mhi[39mst test
test
test test
[38;2;255;40;40mhi[39mst
